go/vt/vtgate/evalengine/evalengine.go

```diff
@@ -312,3 +338,113 @@ func compareNumeric(v1, v2 EvalResult) (int, error) {
 	// v1>v2
 	return 1, nil
 }
+
+func parseDate(expr EvalResult) (t time.Time, err error) {
+	switch expr.typ {
+	case sqltypes.Date:
+		t, err = time.Parse("2006-01-02", string(expr.bytes))
+	case sqltypes.Timestamp, sqltypes.Datetime:
+		t, err = time.Parse("2006-01-02 15:04:05", string(expr.bytes))
+	case sqltypes.Time:
+		t, err = time.Parse("15:04:05", string(expr.bytes))
+		if err == nil {
+			now := time.Now()
+			// setting the date to today's date, because we use AddDate on t
+			// which is "0000-01-01 xx:xx:xx", we do minus one on the month
+			// and day to take into account the 01 in both month and day of t
+			t = t.AddDate(now.Year(), int(now.Month()-1), now.Day()-1)
+		}
```

Comment:
do you have a test case for this? looks like there can be corner cases with timezone (set on MySQL vs VTGate instance)

Reply:
what type of tests do you feel would be helpful here?

Reply:
this is just an internal behavior of Golang where `time.Time`s with no date have their dates set by default to `0000-01-01`

Reply:
filtering on column with time column on left and date column on right.
not sure if evalengine supports that, that would lead the test to come to this code path.

Reply:
the EvalEngine supports it thanks to this code snippet you highlighted. we also have tests for it:
https://github.com/vitessio/vitess/blob/1152eccae4919bacd3436cec151f4b8b3afa0ff8/go/vt/vtgate/evalengine/comparisons_test.go#L804-L815